cou/utils/app_utils.py

```diff
@@ -25,17 +26,24 @@
 logger = logging.getLogger(__name__)
 
 
-async def upgrade_packages(units: Iterable[str], model: COUModel) -> None:
+async def upgrade_packages(
+    units: Iterable[str], model: COUModel, packages_to_hold: Optional[list] = None
+) -> None:
     """Run package updates and upgrades on each unit of an Application.
 
     :param units: The list of unit names where the package upgrade runs on.
     :type units: Iterable[str]
     :param model: COUModel object
     :type model: COUModel
+    :param packages_to_hold: A list of packages to put on hold during package upgrade.
+    :type packages_to_hold: Optional[list]
     :raises RunUpgradeError: When an upgrade fails.
     """
     dpkg_opts = "-o Dpkg::Options::=--force-confnew -o Dpkg::Options::=--force-confdef"
     command = f"apt-get update && apt-get dist-upgrade {dpkg_opts} -y && apt-get autoremove -y"
+    if packages_to_hold:
+        packages = " ".join(packages_to_hold)
+        command = f"sudo apt-mark hold {packages} && {command} ; sudo apt-mark unhold {packages}"
```

Comment:
If we run this command as root, I think we should remove `sudo`

Reply:
Done